src/sql/sqlops.proposed.d.ts

```diff
@@ -50,10 +56,15 @@ declare module 'sqlops' {
 		updateProperties(properties: { [key: string]: any }): Thenable<boolean>;
 	}
 
+	export interface TitledComponent {
```

Comment:
should the interface extend the Component interface?  Or is composition really the correct relationship here?

Reply:
I used this pattern so I can create title for any component. Instead of creating FormInputBox, FormButton, ... I can create InputBox and to add it to a form, I wrap it as FromComponent to set title

---
In reply to: [184465832](https://github.com/Microsoft/sqlopsstudio/pull/1269#discussion_r184465832) [](ancestors = 184465832)